react/src/pages/upload/UploadUtils.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+    DataEntryHeader,
+    DataEntryRow,
+    getDataEntryHeaders,
+    getProp,
+    snakeCaseToTitle,
+} from "../utils";
+
+// Convert a field string (snake_case) into a displayable title (Snake Case)
+export function formatFieldToTitle(field: string): string {
+    return snakeCaseToTitle(field) // convert to title case
+        .replace(/([iI][dD])/g, txt => txt.toUpperCase()); // capitalize any occurrance of "ID"
+}
+
+// Given a DataEntryRow field, return a new DataEntryHeader obj
+export function toColumn(
+    field: keyof DataEntryRow,
+    hidden?: boolean,
+    title?: string
+): DataEntryHeader {
+    return {
+        field: field,
+        title: title
+            ? title
+            : formatFieldToTitle(field),
+        hidden: hidden,
+    };
+}
+
+// Return the specified category of DataEntryHeaders for use in the table
+export function getColumns(category: "required" | "optional" | "RNASeq"): DataEntryHeader[] {
+    return (getProp(getDataEntryHeaders(), category) as Array<keyof DataEntryRow>).map(field =>
```

Comment:
`getDataEntryHeaders()[category]` works, though the typecast is still needed because TypeScript isn't smart enough yet